controller/api/destination/watcher/opaque_ports_watcher.go

```diff
@@ -103,6 +121,12 @@ func (opw *OpaquePortsWatcher) Unsubscribe(id ServiceID, listener OpaquePortsUpd
 			ss.listeners = ss.listeners[:n-1]
 		}
 	}
+
+	if len(ss.listeners) == 0 {
+		delete(opw.subscriptions, id)
```

Comment:
In the Server subscriptions we delete the gauge with `Delete` if there are no listeners. Any reason we aren't doing that here as well?

Reply:
Good catch, it'd better be consistent. I updated in my latest push.